zebra-consensus/src/transaction.rs

```diff
@@ -731,9 +731,28 @@ where
                 ));
             }
 
-            // Consensus rule: The joinSplitSig MUST represent a
-            // valid signature, under joinSplitPubKey, of the
-            // sighash.
+            // # Consensus
+            //
+            // > If effectiveVersion ≥ 2 and nJoinSplit > 0, then:
+            // > - joinSplitPubKey MUST be a valid encoding of an Ed25519 validating key
+            // > - joinSplitSig MUST represent a valid signature under
+            //     joinSplitPubKey of dataToBeSigned, as defined in § 4.11
+            //
+            // https://zips.z.cash/protocol/protocol.pdf#txnconsensus
+            //
+            // The `if` part is indirectly enforced, since the `joinsplit_data`
+            // is only parsed if those conditions apply in
+            // [`Transaction::zcash_deserialize`].
+            //
+            // The valid encoding is defined in
+            //
+            // > A valid Ed25519 validating key is defined as a sequence of 32
+            // > bytes encoding a point on the Ed25519 curve
+            //
+            // https://zips.z.cash/protocol/protocol.pdf#concreteed25519
+            //
+            // which is enforced during signature verification, in both batched
+            // and single verification, when decompressing the encoded point.
```

Comment:
Should we also mention https://zips.z.cash/zip-0215 ?

Reply:
ZIP-215 was incorporated into the spec, in the section linked above, so I don't think it's necessary 

![image](https://user-images.githubusercontent.com/35766/153065061-def593f6-6718-4372-8bb7-e195a4f562d2.png)